frontend/src/components/CouponItem.tsx

```diff
@@ -0,0 +1,110 @@
+import 'bootstrap/dist/css/bootstrap.min.css';
+import { useState } from 'react';
+import { Col, Modal, Row } from 'react-bootstrap';
+import { Link } from 'react-router-dom';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCircleXmark } from '@fortawesome/free-regular-svg-icons';
+import { IconProp } from '@fortawesome/fontawesome-svg-core';
+import CouponItemTemplate from './CouponItemTemplate';
+import TButton from './TButton';
+
+interface CouponItemProps {
+  data: {
+    id: number | null;
+    name: string;
+    policy: string;
+    date: string;
+    introduction: string;
+    tags: { name: string }[];
+  };
+}
+
+const LinkCouponItem = ({ data }: CouponItemProps) => {
+  return (
+    <Link className='none' to={`${window.location.pathname}/${data.id}`}>
+      <div style={{ cursor: 'pointer' }}>
+        <CouponItemTemplate data={data} />
+      </div>
+    </Link>
+  );
+};
+
+const ModalCouponItem = ({ data }: CouponItemProps) => {
+  const [show, setShow] = useState(false);
+  const handleShow = () => setShow(true);
+  const handleClose = () => setShow(false);
+  return (
+    <>
+      <div style={{ cursor: 'pointer' }} onClick={handleShow}>
+        <CouponItemTemplate data={data} />
+      </div>
+      <Modal show={show} onHide={handleClose} centered className='coupon_modal'>
+        <Modal.Header>
+          <div className='right' style={{ width: '100%' }}>
+            <FontAwesomeIcon icon={faCircleXmark as IconProp} size='2x' onClick={handleClose} />
+          </div>
+        </Modal.Header>
+        <Modal.Body>
+          <Row>
+            <Col xs={3} md={2} className='center'>
+              <img src='../../images/person.png' className='user_img' />
```

Comment:
currently i put images in /public all images are from there, so do you mean that we remove the images from /public and create /assets in src then include it?
i thought we will get data from backend so the /images in public is just temporary

Reply:
if the images are user uploaded then yes it will come from the backend, if it is just site decorations then it's fine to leave it at the frontend and serve it from the static file server

Reply:
because from what I've known, minio's image urls are not permanent

Reply:
Ill change the site-provide img into ``import Image from '@assets/image.png'``. (banner in main page, bg in sign in, etc)
And leave some dynamic img as it is. (product pic, user icon, etc)